R/slurm_utils.R

```diff
@@ -48,3 +49,22 @@ local_slurm_array <- function(slr_job) {
         system(paste(rscript_path, "--vanilla local_run.R"))
     }, finally = setwd(olddir))
 }
+
+# Submit job capturing jobid
+submit_slurm_job <- function(tmpdir, jobid) {
+    old_wd <- setwd(tmpdir)
+    tryCatch({
+        submission <- system("sbatch submit.sh", intern = TRUE)
+        cat(submission, sep = '\n')
+        jobid <- regmatches(submission, regexpr('\\d*$', submission))
+    }, finally = setwd(old_wd))
+    return(as.integer(jobid))
+}
+
+# Submit a dummy job with dependency to block R script
+wait_for_job <- function(slr_job) {
+    srun <- sprintf(
+        'srun -n1 -t0:1 -o/dev/null -Q -dafterany:%d /bin/hostname',
```

Comment:
This will work, however may want to use --dependency= as there is a discrepency between the llnl docs say (-P) https://computing.llnl.gov/tutorials/linux_clusters/man/srun.txt and what srun on our cluster accepts (-d).